File: Longest Palindromic Substring/ExpandAroundCenter.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        start = 0
        end = 0
        if(s is None) or (len(s) <= 0):
            return ""
        for i in range(len(s)):
            len1 = self.expandAroundCenter(s, i, i)
            len2 = self.expandAroundCenter(s, i, i+1)
            length = max(len1, len2)
            if length > end - start:
                start = i - (length-1)/2
                end = i + length/2
        
        return s[start: end+1]

    def expandAroundCenter(self, s, left, right):
        l = left
        r = right
        while l >= 0 and r < len(s) and s[l] == s[r]:
            l -= 1
            r += 1

        return r-l-1

File: Longest Palindromic Substring/ExpandAroundCenter.java
class Solution {
    public static  String longestPalindrome(String s) {
			int start = 0, end = 0;

			if(s == null || s.length() < 1)
				return "";
				
				for (int i = 0; i < s.length(); i++) {
					int len1 = expandAroundCenter(s, i, i);
					int len2 = expandAroundCenter(s, i, i+1);
					int len = Math.max(len1, len2);
					if(len > end-start) {
						start = i-(len-1)/2;
						end = i + len/2;
					}					
				}
			return s.substring(start, end+1);
			
	}
		private static int expandAroundCenter(String s, int lift, int right) {
			int L = lift;
			int R = right;
			while(L >= 0 && R < s.length() && s.charAt(L) == s.charAt(R)) {
				L--;
				R++;
			}
			
			return R-L-1;
		}
}
